docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+        - name: TEMPORAL_NAMESPACE
+          value: "production"
+        # No versioning environment variables
+```
+
+**Current Deployment Process:**
+1. Build new worker image
+2. Update Deployment with new image
+3. Kubernetes rolls out new pods, terminating old ones
+4. **Risk**: Running workflows may fail if code changes break compatibility
+
+### After: Versioned Workers with Controller
+
+```yaml
+# Single CRD manages multiple versions automatically
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: my-worker
+spec:
+  replicas: 3
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: production
+  cutover:
+    strategy: Progressive  # Gradual rollout of new versions
+    steps:
+      - rampPercentage: 10
+        pauseDuration: 5m
+      - rampPercentage: 50
+        pauseDuration: 10m
+  sunset:
+    scaledownDelay: 1h
+    deleteDelay: 24h
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: my-worker:v1.2.4  # Update this to deploy new versions
+        # Note: Controller automatically adds versioning environment variables:
+        # TEMPORAL_HOST_PORT, TEMPORAL_NAMESPACE, TEMPORAL_DEPLOYMENT_NAME, WORKER_BUILD_ID
+```
+
+**New Deployment Process:**
+1. Build new worker image  
+2. Update `TemporalWorkerDeployment` CRD with new image
+3. Controller creates new Kubernetes `Deployment` for the new version
+4. Controller gradually routes new workflows to new version
+5. Old version continues handling existing workflows until they complete
+6. **Safety**: No disruption to running workflows, automated rollout control
+
+**Key Benefits:**
+- ✅ **Zero-disruption deployments** - Running workflows continue on original version
+- ✅ **Automated version management** - Controller handles registration and routing
+- ✅ **Progressive rollouts** - Gradual traffic shifting with automatic pause points
+- ✅ **Easy rollbacks** - Instantly route new workflows back to previous version
+- ✅ **Workflow continuity** - Deterministic execution preserved across deployments
+
+## Migration Strategy
+
+### Safe Migration Approach
+
+The migration from unversioned to versioned workflows requires careful planning to avoid disrupting running workflows. The key is to transition gradually while maintaining workflow continuity.
+
+**Key Principles:**
+- **Start with Manual strategy** - Prevents automatic promotions during initial setup
+- **Migrate one worker deployment at a time** - Reduces risk and allows learning
+- **Test thoroughly in non-production** - Validate the approach before production migration
+- **Preserve running workflows** - Ensure in-flight workflows complete successfully
+- **Enable automation gradually** - Move to Progressive rollouts only after validation
+
+### Migration Phases
+
+#### Phase 1: Preparation
+1. **Install the controller** in non-production environments
+2. **Update worker code** to support versioning (if needed)
```

Comment:
the "if needed" feels weird here given that we said that the workers were unversioned as a requirement